src/leappto/cli.py

```diff
@@ -122,6 +140,219 @@ def _find_machine(ms, name):
                 return machine
         return None
 
+    class PortScanException(Exception):
+        pass
+    
+    class PortCollisionException(Exception):
+        pass
+
+    class PortMap(OrderedDict):
+        PROTO_TCP = "tcp"
+        PROTO_UDP = "udp"
+
+        def __init__(self, port_map = None):
+            super(PortMap, self).__init__()
+            self.__init_structure()
+
+            if port_map and isinstance(port_map, OrderedDict):
+                self.__from_dict(port_map)
+            elif port_map:
+                raise AttributeError("Unknown port_map type")
+
+
+        def __init_structure(self):
+            self[self.PROTO_TCP] = OrderedDict()
```

Comment:
Basically we do not, unless we want to keep order for user output. Which we actually want to for port inspect.